pkg/controller/credentialsrequest/credentialsrequest_controller.go

```diff
@@ -274,28 +277,45 @@ func (r *ReconcileCredentialsRequest) Reconcile(request reconcile.Request) (reco
 		logger.Debug("found secret namespace")
 	}
 
-	// FIXME / TODO: check for proper secret annotation
+	// Now figure out whether we need to create new creds/secrets to satisfy the CredentialsRequest
 
-	// Hand over to the actuator:
-	exists, err := r.Actuator.Exists(context.TODO(), cr)
+	updateNeeded, err := r.Actuator.NeedsUpdate(context.TODO(), cr)
```

Comment:
This will catch things like the access key being deleted in AWS, and the policy modified in AWS correct?

What would happen if we removed permissions, or if someone went to AWS and manually granted that access key elevated privs? My suspicion is that because we can still do what is necessary, the code will never correct the problem. If so I fear that needs fixing, teams that will use this will need to iterate on the perms they need, if they remove one they're going to expect it gets removed in AWS, and not doing so is going to cause bugs where they may think they didn't need it but actually did. Can we re-incorporate comparison of policy documents somewhere?

Reply:
will change to actually check the policy docs